bazel/envoy_build_system.bzl

```diff
@@ -652,54 +150,41 @@ def envoy_proto_descriptor(name, out, srcs = [], external_deps = []):
         tools = ["//external:protoc"],
     )
 
-# Selects the given values if hot restart is enabled in the current build.
-def envoy_select_hot_restart(xs, repository = ""):
-    return select({
-        repository + "//bazel:disable_hot_restart_or_apple": [],
-        "//conditions:default": xs,
-    })
-
-# Selects the given values if Google gRPC is enabled in the current build.
-def envoy_select_google_grpc(xs, repository = ""):
-    return select({
-        repository + "//bazel:disable_google_grpc": [],
-        "//conditions:default": xs,
-    })
-
 # Dependencies on Google grpc should be wrapped with this function.
 def envoy_google_grpc_external_deps():
-    return envoy_select_google_grpc([_envoy_external_dep_path("grpc")])
+    return envoy_select_google_grpc([envoy_external_dep_path("grpc")])
 
 def envoy_select_boringssl(if_fips, default = None):
     return select({
         "@envoy//bazel:boringssl_fips": if_fips,
         "//conditions:default": default or [],
     })
 
-# Select the given values if default path normalization is on in the current build.
-def _envoy_select_path_normalization_by_default_internal(xs, repository = ""):
-    return select({
-        repository + "//bazel:enable_path_normalization_by_default": xs,
-        "//conditions:default": [],
-    })
-
-def _envoy_select_perf_annotation_internal(xs):
-    return select({
-        "@envoy//bazel:enable_perf_annotation": xs,
-        "//conditions:default": [],
-    })
-
-# Select the given values if exporting is enabled in the current build.
-def _envoy_select_exported_symbols_internal(xs):
-    return select({
-        "@envoy//bazel:enable_exported_symbols": xs,
-        "//conditions:default": [],
-    })
-
-def _envoy_select_force_libcpp_internal(if_libcpp, default = None):
-    return select({
-        "@envoy//bazel:force_libcpp": if_libcpp,
-        "@envoy//bazel:apple": [],
-        "@envoy//bazel:windows_x86_64": [],
-        "//conditions:default": default or [],
-    })
+# Here we create wrappers for each of the public targets within the separate bazel
+# files loaded above. This is necessary so that consumers who had previously used
+# macros imported from envoy_build_system.bzl don't have to change their BUILD
+# files to reference the new location.
```

Comment:
I think it's fine to keep this as the permanent abstraction layer, i.e. a single import of `envoy_build_system.bzl`. The rationale is that we don't want to create too much reliance on the internal structure of the `.bzl` file elsewhere, and the surface of exported macros is small enough that we don't need to have fine grained `.bzl` disaggregation. If that sounds reasonable, maybe update the comments here to reflect.

Reply:
Sounds good, done. I've updated the comments throughout the files to instruct readers to only `load()` `envoy_build_system.bzl`. We can also prepend envoy_library.bzl, envoy_binary.bzl, etc. with an `_` to further discourage their use.